keras_nlp/utils/python_utils.py

```diff
@@ -11,9 +11,21 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Utilities for models and preprocessors."""
+"""Utilities with miscellaneous python extensions."""
 
 
 class classproperty(property):
+    """Define a class level property."""
+
     def __get__(self, _, owner_cls):
         return self.fget(owner_cls)
+
+
+def format_docstring(**replacements):
```

Comment:
Since this delegates to `format` you might want to go with the `format()` signature. No strong opinion though. https://docs.python.org/3/library/stdtypes.html#str.format

Reply:
Good idea.

Reply:
Actually given that we are doing our own double bracket format anyway, will not support the *args part. We don't really want to have people doing `{{0}}` anyway, it's not super readable.